megamek/src/megamek/server/totalwarfare/MovementHandler.java

```diff
@@ -12,30 +30,20 @@
 import java.util.*;
 import java.util.stream.Collectors;
 
-class MovementHandler extends AbstractTWRuleHandler {
-    
-    private final Entity entity;
-    private final MovePath md;
-    private final Map<UnitTargetPair, LosEffects> losCache;
-
-    /**
-     * Steps through an entity movement packet, executing it.
-     *
-     * @param gameManager The server's GameManager
-     * @param entity   The Entity that is moving
-     * @param md       The MovePath that defines how the Entity moves
-     * @param losCache A cache that stores Los between various Entities and
-     *                 targets.  In double blind games, we may need to compute a
-     *                 lot of LosEffects, so caching them can really speed
-     *                 things up.
-     */
-    MovementHandler(TWGameManager gameManager, Entity entity, MovePath md, Map<UnitTargetPair, LosEffects> losCache) {
-        super(gameManager);
-        this.entity = entity;
-        this.md = md;
-        this.losCache = (losCache == null) ? new HashMap<>() : losCache;
-    }
-    
+/**
+ * Steps through an entity movement packet, executing it.
+ *
+ * @param gameManager The server's GameManager
+ * @param entity   The Entity that is moving
+ * @param md       The MovePath that defines how the Entity moves
+ * @param losCache A cache that stores Los between various Entities and
+ *                 targets.  In double blind games, we may need to compute a
+ *                 lot of LosEffects, so caching them can really speed
+ *                 things up.
+ */
+record MovementHandler(TWGameManager gameManager, Entity entity, MovePath md, Map<UnitTargetPair, LosEffects> losCache)
```

Comment:
Making this a record prevents the use of member fields, which are necessary for breaking the code down into smaller chunks.